packages/blade/src/components/Typography/Heading/__tests__/Heading.native.test.tsx

```diff
@@ -13,43 +13,43 @@ describe('<Heading />', () => {
     expect(toJSON()).toMatchSnapshot();
   });
 
-  it('should render Heading with variant "small" and contrast "high"', () => {
+  it('should render Heading with size "small" and contrast "high"', () => {
```

Comment:
If there is a possibility of impossible state, let's test that unhappy path.

Reply:
Pretty simple component so can't think of any impossible state as such 🤔 

One case would be having `size` on `variant="subheading"` but TS would handle that right? do we need runtime checks there?

Reply:
Added the runtime check and unhappy flow for size and subheading variant

Reply:
Cool. 
IMO we should all together try to reduce the number of impossible states as much possible. It's always better DX even with TS.